torch/testing/_internal/distributed/multi_threaded_pg.py

```diff
@@ -61,6 +62,16 @@ def bitwise_reduce(tensors, op):
     ReduceOp.BXOR: partial(bitwise_reduce, op=torch.bitwise_xor),
 }
 
+class AllToAll:
+    def work(self, data):
+        world_size = len(data)
+        for dest_rank in range(world_size):
+            output_tensor_list, _ = data[dest_rank]
+            for src_rank in range(world_size):
+                _, input_tensor_list = data[src_rank]
+                with torch.no_grad():
```

Comment:
nit: If you want to avoid the overhead of entering and exiting `torch.no_grad()` context, you can decorate the function with `@torch.no_grad()`. This might be good to do since the number of enters/exits is quadratic in world size.